digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala

```diff
@@ -1020,4 +1098,300 @@ class TrafficSignParkingProhibitionGenerator(roadLinkServiceImpl: RoadLinkServic
+
+    val relevantLink = relevantLinkOnChain(roadLinks, trafficSigns)
+    val newSegments = relevantLink.flatMap { case (roadLink, startPointOfInterest, lastPointOfInterest) =>
+      baseProcess(trafficSigns, roadLinks, roadLink, (startPointOfInterest, lastPointOfInterest, None), Seq())
+    }.distinct
+
+  val mergedSegments = newSegments.map { x =>
+                          val newProps = if (existingSegments.nonEmpty) withdraw( existingSegments.head.value, x.value )
+                                          else x.value
+
+                          TrafficSignToLinear(x.roadLink, newProps, x.sideCode, x.startMeasure, x.endMeasure, x.signId, x.oldAssetId )
+                      }
+
+    val groupedAssets = mergedSegments.groupBy(_.roadLink)
+    val assets = fillTopology(roadLinks, groupedAssets)
+
+    convertRoadSegments(assets, findStartEndRoadLinkOnChain(roadLinks)).toSet
+  }
+
+  override def convertRoadSegments(segments: Seq[TrafficSignToLinear], endRoadLinksInfo: Seq[(RoadLink, Option[Point], Option[Point])]): Seq[TrafficSignToLinear] = {
+    segments
+  }
+
+  def generateSegmentPiece(currentRoadLink: RoadLink, sign: PersistedTrafficSign, value: Value, endDistance: Option[Double], direction: Int): TrafficSignToLinear = {
+    if (debbuger) println("generateSegmentPiece")
+    endDistance match {
+      case Some(mValue) =>
+        if (currentRoadLink.linkId == sign.linkId) {
+          val orderedMValue = Seq(sign.mValue, mValue).sorted
+
+          TrafficSignToLinear(currentRoadLink, value, SideCode.apply(sign.validityDirection), orderedMValue.head, orderedMValue.last, Set(sign.id))
+        } else {
+          val (starMeasure, endMeasure) = if (SideCode.apply(direction) == TowardsDigitizing)
+            (0.toDouble, mValue)
+          else {
+            val length = GeometryUtils.geometryLength(currentRoadLink.geometry)
+            (length - mValue, "%.3f".formatLocal(java.util.Locale.US, length).toDouble)
+          }
+          TrafficSignToLinear(currentRoadLink, value, SideCode.apply(direction), starMeasure, endMeasure, Set(sign.id))
+        }
+      case _ =>
+        if (currentRoadLink.linkId == sign.linkId) {
+          val (starMeasure, endMeasure) = if (SideCode.apply(direction) == AgainstDigitizing)
+            (0L.toDouble, sign.mValue)
+          else {
+            val length = GeometryUtils.geometryLength(currentRoadLink.geometry)
+            (sign.mValue, "%.3f".formatLocal(java.util.Locale.US, length).toDouble)
+          }
+
+          TrafficSignToLinear(currentRoadLink, value, SideCode.apply(direction), starMeasure, endMeasure, Set(sign.id))
+        }
+        else {
+
+          val length = GeometryUtils.geometryLength(currentRoadLink.geometry)
+          TrafficSignToLinear(currentRoadLink, value, SideCode.apply(direction), 0, "%.3f".formatLocal(java.util.Locale.US, length).toDouble, Set(sign.id))
+        }
+    }
+  }
```

Comment:
This method's totally equal with other already existing at TrafficSignParkingProhibitionGenerator class.

Since "TrafficSignParkingProhibitionGenerator" and "TrafficSignRoadWorkGenerator" extend from "TrafficSignDynamicAssetGenerator" let's try to put the method on father class and using on the sons.

Reply:
After talking, changes made as requested